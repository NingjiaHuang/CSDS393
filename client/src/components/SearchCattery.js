import React, {Fragment, useState, useEffect} from 'react'
import mock_data from "./MOCK_DATA.json"
import { Button } from 'react-bootstrap';

const SearchCattery = ({setAuth}) =>{
  const[searchTerm, setSearchTerm] = useState("")
  const[catteries, setCatteries] = useState([]);
  const getCatteries = async() => {
      try{
          
          const response = await fetch("http://localhost:4020/api/v1/catteries")
          const jsonData = await response.json()
          setCatteries(jsonData);

      } catch (err){
          console.error(err.message);
      }
  }

  useEffect(() =>{
      getCatteries();
  }, [])

    return(
        <Fragment>
             {" "}
          <input type="text" className="form-control" 
          data-testid ="searchcattery"
          placeholder="search cattery here" 
          onChange={event => {setSearchTerm(event.target.value)}}
          />
          <table class="table mt-5 text-center">
              <thead>
                  <tr>
                      <th>Cattery Name</th>
                      <th>Owner Name</th>
                  </tr>
              </thead>
              <tbody>
              {catteries.filter((val) =>{  //if get all catteries completed, change "mock_data" to catteries
                  if (searchTerm == ""){
                    return val
                  } else if (val.cattery_name.toLowerCase().includes(searchTerm.toLowerCase()) ){
                    return val
                  }
              }
              ).map((val, key) => (
                  <tr key={key}>
                      <td>{val.cattery_name}</td>
                      <td>{val.owner_name}</td>
                      <td>
                      </td>
                  </tr>
              ))
              }
            </tbody>
        </table>
        </Fragment>
    )
}

export default SearchCattery;