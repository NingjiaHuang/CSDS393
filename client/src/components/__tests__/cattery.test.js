import {render, screen, cleanup, getByTestId, fireEvent} from '@testing-library/react';
import {mount} from 'enzyme';
import SearchCattery from '../SearchCattery';

import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';

afterEach(() =>{
    cleanup();
})


test('matches snapshot of search cattery', () =>{
    const tree = renderer.create(<SearchCattery/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("search box should accept text", () => {
    const {getByTestId} = render(<SearchCattery/>);
    const searchInputNode = getByTestId('searchcattery');
    expect(searchInputNode.value).toMatch("");
    fireEvent.change(searchInputNode, {target:{value:'test@test.com'}});
    expect (searchInputNode.value).toMatch("test@test.com")
})