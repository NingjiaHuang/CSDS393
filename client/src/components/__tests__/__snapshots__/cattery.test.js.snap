// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of search cattery 1`] = `
Array [
  " ",
  <input
    className="form-control"
    data-testid="searchcattery"
    onChange={[Function]}
    placeholder="search cattery here"
    type="text"
  />,
  <table
    class="table mt-5 text-center"
  >
    <thead>
      <tr>
        <th>
          Cattery Name
        </th>
        <th>
          Owner Name
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
